solr/core/src/java/org/apache/solr/metrics/SolrMetricsContext.java

```diff
@@ -116,21 +134,94 @@ public SolrMetricsContext getChildContext(Object child) {
    * Register a metric name that this component reports. This method is called by various metric
    * registration methods in {@link org.apache.solr.metrics.SolrMetricManager} in order to capture
    * what metric names are reported from this component (which in turn is called from {@link
-   * org.apache.solr.metrics.SolrMetricProducer#initializeMetrics(SolrMetricsContext, String)}).
+   * SolrMetricProducer#initializeMetrics(SolrMetricsContext,
+   * io.opentelemetry.api.common.Attributes, String)}).
    */
+  // TODO We can continue to register metric names
   public void registerMetricName(String name) {
     metricNames.add(name);
   }
 
   /** Return a snapshot of metric values that this component reports. */
+  // TODO Don't think this is needed anymore
   public Map<String, Object> getMetricsSnapshot() {
     return MetricUtils.convertMetrics(getMetricRegistry(), metricNames);
   }
 
+  public LongCounter longCounter(String metricName, String description) {
```

Comment:
Seeing all these simply pass through to metricManager, I wonder if you plan to remove SolrMetricsContext once DropWizard is gone?  Or maybe it stays but you have yet to introduce OTEL scope, so it shows up here?